middleware/kubernetes/kubernetes.go

```diff
@@ -67,9 +70,8 @@ func (k *Kubernetes) Lookup(state request.Request, name string, typ uint16) (*dn
 }
 
 // IsNameError implements the ServiceBackend interface.
-// TODO(infoblox): implement!
 func (k *Kubernetes) IsNameError(err error) bool {
-	return false
+	return err == noItemsErr || err == nsUnexposedErr
```

Comment:
Well - no... since we are doing the service matching on our side, k8s has no knowledge of whether or not a service matched our query.